lib/x11utils.pm

```diff
@@ -223,6 +223,8 @@ sub handle_login {
     assert_screen 'displaymanager-password-prompt';
     type_password;
     send_key 'ret';
+    assert_screen([qw(generic-desktop gnome-activities opensuse-welcome-gnome40-activities opensuse-welcome)], 60);
+    send_key('esc') if match_has_tag('gnome-activities') || match_has_tag('opensuse-welcome-gnome40-activities');
 }
```

Comment:
It might take a while until leaving the activities view is complete, so I suggest

`assert_screen('generic-desktop') unless match_has_tag('generic-desktop');`

Reply:
Thanks for your suggestion and the code has been updated.

Reply:
The change doesn't look correct to me. What I had in mind is:

```
    assert_screen([qw(generic-desktop gnome-activities)], 60);
    send_key('esc') if match_has_tag('gnome-activities');
    assert_screen('generic-desktop') unless match_has_tag('generic-desktop');
```

Reply:
I agree with 
`assert_screen([qw(generic-desktop gnome-activities)], 60);
 send_key('esc') if match_has_tag('gnome-activities');`

But, I don't think we need `assert_screen('generic-desktop') unless match_has_tag('generic-desktop');`
I checked and found almost all the needles that contain the tag "gnome-activities" also contain the tag "generic-desktop".
The only exception is opensuse-welcome-gnome40-20210603.* which probably outdated since we now have a newer one needle with the same name but a different date opensuse-welcome-gnome40-20210619.* (it doesn't have the tag "gnome-activities")

Reply:
> I checked and found almost all the needles that contain the tag "gnome-activities" also contain the tag "generic-desktop".

Yes, but that is actually broken and needs to be changed. `generic-desktop` and `gnome-activities` are vastly different states of the desktop, so a needle containing both just makes no sense.

Without that change, it's not easily possible to tell for sure whether the system got out of activities view. You'd have to match `generic-desktop` repeatedly until the result no longer contains the `gnome-activities` tag, which is backwards.

All of the occurences of `generic-desktop` assume that the system is not in activities view, because it needs special treatment.

Reply:
So, we still need `assert_screen('generic-desktop') unless match_has_tag('generic-desktop');`. And in the meanwhile, we'd better delete the needles that contain both of the two tags `generic-desktop` and `gnome-activities`.
Am I right?

Reply:
Yep, that's how my idea how to deal with the activities view looks like. Instead of just deleting the needles, just removing the `generic-desktop` tag from them would save reneedling.

If you agree with the approach, you could do that in a PR to the needles repo, used in the VRs here as well.

Reply:
Thanks for your patience and help. 
Needles and codes have been updated.